test/node/client-deselect.js

```diff
@@ -0,0 +1,109 @@
+import fixtures from 'webtorrent-fixtures'
+import MemoryChunkStore from 'memory-chunk-store'
+import test from 'tape'
+import WebTorrent from '../../index.js'
+
+function setupClient ({ t, onTorrent, onDone, addTorrentProps = {} }) {
+  const client1 = new WebTorrent({ dht: false, tracker: false, lsd: false })
+  client1.on('error', function (err) { t.fail(err) })
+  client1.on('warning', function (err) { t.fail(err) })
+
+  const client2 = new WebTorrent({ dht: false, tracker: false, lsd: false })
+  client2.on('error', function (err) { t.fail(err) })
+  client2.on('warning', function (err) { t.fail(err) })
+
+  const parsedTorrent = Object.assign({}, fixtures.leaves.parsedTorrent)
+
+  client1.seed(fixtures.leaves.content, {
+    name: 'Leaves of Grass by Walt Whitman.epub',
+    announce: [],
+    store: MemoryChunkStore
+  }, () => {
+    client2.add(parsedTorrent, { store: MemoryChunkStore, ...addTorrentProps }, (torrent) => {
+      onTorrent(torrent)
+
+      torrent.addPeer(`localhost:${client1.torrentPort}`)
+
+      torrent.once('done', () => {
+        onDone(torrent)
+
+        client1.destroy(function (err) { t.error(err, 'client1 destroyed') })
+        client2.destroy(function (err) { t.error(err, 'client2 destroyed') })
+      })
+    })
+  })
+}
+
+test('client.select: whole torrent', function (t) {
+  t.plan(3)
+
+  setupClient({
+    t,
+    onTorrent: (torrent) => {
+      torrent.select(0, torrent.pieces.length - 1)
+    },
+    onDone: (torrent) => {
+      t.equal(torrent.pieces.filter((a) => a === null).length, torrent.pieces.length)
+    }
+  })
+})
+
+test('client.select: partial torrent', function (t) {
+  t.plan(3)
+
+  let lastPiece
```

Comment:
this is touch missleading, torrent's have a concept of "lastPiece" which is "the last piece in the torrent which doesn't have the length of a normal piece, instead is the remaining data" so this could be treated as lastPieceLength, better be safe and rename this to `lastPieceIndex` everywhere in this test file

Reply:
didn't know about that, plus this test is not originally mine. thanks for the info. renamed them all